### ISSUE TITLE
removePathAccents true leave dot and forward slash in permalinks

### ISSUE THREAD

**Issue by frjo** [spf13/hugo#3577]
I have this permalink set up for posts.

    permalinks:
      post: "/post/:year/:month/:day/:slug"

If no slug is preset the post title will be used instead, and that is so for more or less all my posts.

Together with "removePathAccents: true" I get a problem with characters like a dot or forward slash in the posts title that does not get replaced in the generated permalink for the post. I have solved it for now by adding a slug to the post with problematic titles.

This example title

    Lörem ipsum dölor sit amet/consectetur do example.org

generates the following permalink

    /post/2006/02/02/lorem-ipsum-dolor-sit-amet/consectetur-do-example.org/

I had hoped it would become

    /post/2006/02/02/lorem-ipsum-dolor-sit-amet-consectetur-do-example-org/

If I set "removePathAccents: false" I get

    /post/2006/02/02/lörem-ipsum-dölor-sit-amet-consectetur-do-example-org/

but I would prefer to have åäö replaced with ago in permalink.

**Comment by bep** [comment]
You should create an issue about **one** thing.

**Comment by frjo** [comment]
Better now? The removePathAccents part was due to bad testing by me, it has nothing to do with the problem.

**Comment by tsuereth** [comment]
I've recently been working on a similar issue (for taxonomy terms), and while I was in there, I put in a fix for slashes in titles.  See #4091 ; for your example, this will generate-

```
/post/2006/02/02/lorem-ipsum-dolor-sit-amet-consectetur-do-example.org/
```

My changes don't do anything with dots -- for my hugo site, the dots are fine.  But, it seems to me like it could make sense to add a configurable flag like "disallowDotsInPathSegments" (or, probably a shorter flag name) to optionally alter that behavior.

Does this sound right to you?  If my PR is accepted, I could follow it up with something like that.

**Comment by moorereason** [comment]
@frjo,
You raise two issues:

1. **Forward slashes**: Agreed.  Similar to #4090.  Can likely solve at the same time.

1. **Dots**: I see no reason to treat a dot as a special case.  They have no special, reserved meaning in file system paths or URLs.

I'm renaming this issue to focus on the forward slash problem.  If you want to argue for the dots escaping, open a separate issue.

**Comment by tcg** [comment]
Apologies if I shouldn't "me, too", but just noticed this in `0.42.2` and `0.43`. I originally ran across it because of hash marks, which I fully expected to be escaped, which I detailed here: https://discourse.gohugo.io/t/hash-pound-sign-not-escaped-in-title-permalink-breaking-urls/12825

I was asking there if I should open a separate issue, but was open to the possibility that I am just doing something wrong. Glad to document and file a bug if necessary.

**Comment by max-arnold** [comment]
Could anyone test this again with Hugo 0.54.0?  After #5571 was merged, slashes in titles are not escaped anymore (at least for me).  I think this issue should be reopened 😞

**Comment by sgumirov** [comment]
I can confirm regression with slash is not being escaped in url. Any config flags to enable this?

**Comment by moorereason** [comment]
This issue has been closed for months.  Please open a new issue.
